### ISSUE TITLE
Scheduler: support for cancelable scripts

### ISSUE THREAD

**Issue by kdbo** [jsreport/jsreport#101]
Hi,

I have a report that needs to be generated via the scheduler.
This script should generate a report, only for the next working day but not if that next day is a bank holiday.

The first part is easy, we just run the script from monday to friday via the scheduler.
then we request the data in the beforeRender function from the REST API.
The REST API will return data for the next businessday, OR a cancelRequest if next BusinessDay is a bankholiday
So I thought I needed to use request.cancel function.
But that still throws an error, while I think, it should not throw an error, but just marks this report run as canceled.

````javascript
function beforeRender(done) {
  var uri = request.data.api.uri;
  require('request').post(
      { 
        url: uri + '/api/getNextBDdata', 
        json:true,
        strictSSL : false,
        headers: {'content-type' : 'application/x-www-form-urlencoded'},
        body:    "apikey="+request.data.api.key,
      }, function(err, response, body) {
    if(body.cancelRequest) { // My API detected a bankholiday, thus it has sent me a cancelRequest
        console.log('beforeRender got cancelRequest!');
        // Next step should not result in an error but silently be dropped, 
        // while the schedule run should be marked as cancelled, not error
        request.cancel(body.additionalInfo);
        return;
    }
    if(body.error) {
       done(body.error);
       return; 
   }
    request.data.resultSet = body;
    done();
  });
}
````

**Comment by pofider** [comment]
Is there any other benefit for this than a different representation in the UI for Schedule detail?

You could call `done(new Error("Job cancelled"));` to error the job. Then the job is marked as error but you at least get a meaningful message when clicking on the state.

I am just trying to estimate a priority for this...

**Comment by pofider** [comment]
I totally forget `request.cancel` is actually supported in scripts so you get the meaningful message anyway. Please ignore what I wrote with `done(new Error("Job cancelled"));`

**Comment by kdbo** [comment]
Hi Jan,

Visual confirmation was the main goal. So no real urgency as far as I am concerned.
I have already better visibility with the jsreport scheduler vs real cron jobs :-)
I was also thinking about sending mails when a (real) error occurs, but I might do that at the same place in the script where I decide to cancel a job or not, so no need for that either.

**Comment by pofider** [comment]
We have this implemented now, it should be in the next release. I will close this when it is published.

**Comment by pofider** [comment]
Changes released in 0.12.0
